### DISCUSSION TITLE
gpu rendering in vcl

### DISCUSSION THREAD

**Discussion by mahmoudikia** [skia4delphi/skia4delphi#185] [Q&A] [answered]
hi
how can i enable gpu rendering in vcl mode?

thanks.

**Comment by viniciusfbb** [comment]
Hello. It's not very simple. Skia does not create or initialize backend graphics libraries (OpenGl, Vulkan, DirectX, Metal, ...). This was done to give the developer/framework more freedom to create and initialize the backend as they wish, and then pass the context of the backend to Skia internally to draw using it. I think this works correctly because, in fact, there are several situations where this flexibility is great, but on the other hand you must have some familiarity with the backend to do the initialization correctly.

Currently, we implement the creation and initialization of the OpenGL and Metal backends, but only in FMX, to replace the forms renderer with our own renderer (ie our own TCanvas) based on Skia.

In the future, we intend to create some VCL backends for Skia4Delphi controls, especially for TSkPaintBox and TSkAnimatedPaintBox. There is already an open issue for this feature: #108.

**Comment by viniciusfbb** [comment, marked as answer]
We have added the GPU render to Vcl on TSkAnimatedPaintBox control in version [5.0.0+](https://github.com/skia4delphi/skia4delphi/releases), that you can enable by setting the property TSkAnimatedPaintBox.RenderBackend to TSkControlRenderBackend.HardwareAcceleration.
